src/analysis/inner_product_space/l2_space.lean

```diff
@@ -437,6 +465,47 @@ hilbert_basis.coe_mk hv _
 
 omit hv
 
+-- Note : this should be `b.repr` composed with an identification of `lp (λ i : ι, 𝕜) 2` with
+-- `pi_Lp 2 (λ i : ι, 𝕜)`, but we don't have that yet (July 2022).
+/-- An orthonormal basis is an Hilbert basis. -/
+protected def _root_.orthonormal_basis.to_hilbert_basis [fintype ι] (b : orthonormal_basis ι 𝕜 E) :
+  hilbert_basis ι 𝕜 E :=
+hilbert_basis.mk b.orthonormal
+begin
+  rw [← set.image_univ, ← finset.coe_univ, ← finset.coe_image],
+  have := (span 𝕜 (finset.univ.image b : set E)).closed_of_finite_dimensional,
+  rw [this.submodule_topological_closure_eq, finset.coe_image, finset.coe_univ, set.image_univ,
+      ← orthonormal_basis.coe_to_basis],
+  exact b.to_basis.span_eq
+end
+
+@[simp] lemma _root_.orthonormal_basis.coe_to_hilbert_basis [fintype ι]
+  (b : orthonormal_basis ι 𝕜 E) : (b.to_hilbert_basis : ι → E) = b :=
+hilbert_basis.coe_mk _ _
+
+protected lemma _root_.orthonormal_basis.orthogonal_projection_eq_sum [fintype ι]
+  {U : submodule 𝕜 E} [complete_space E] [complete_space U] (b : orthonormal_basis ι 𝕜 U) (x : E) :
+  (orthogonal_projection U x) = ∑ i, ⟪(b i : E), x⟫ • b i :=
+begin
```

Comment:
Could you please make the proof directly using `orthonormal_basis` rather than by passing to `hilbert_basis`, and have it live in the `pi_L2` file?